src/models/OTPValidate.model.ts

```diff
@@ -0,0 +1,99 @@
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed
+ on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Kevin Leyow <kevin.leyow@modusbox.com>
+ --------------
+ ******/
+
+import {
+  thirdparty as tpAPI
+} from '@mojaloop/api-snippets'
+import { resolve } from 'path'
+import { A2SModelConfig } from './a2s.model'
+import { KVS } from '../shared/kvs'
+import { PubSub, Message } from '../shared/pub-sub'
+import { ThirdpartyRequests, Logger as SDKLogger } from '@mojaloop/sdk-standard-components'
+import { OutboundOTPValidateResponse } from './OTPValidate.interface'
+
+export interface OTPValidateModelArgs {
+  consentRequestId: string,
+  fspId?: string,
+  consentRequest?: tpAPI.Schemas.ConsentRequestsIDPatchRequest
+}
+
+export class OTPValidateModelConfig implements A2SModelConfig<OTPValidateModelArgs, OutboundOTPValidateResponse> {
+  public readonly key: string
+  public readonly kvs: KVS
+  public readonly logger: SDKLogger.Logger
+  public readonly pubSub: PubSub
+  public readonly modelName = 'OTPValidate'
+  public readonly thirdpartyRequests: ThirdpartyRequests
+  public readonly requestProcessingTimeoutSeconds = 10000
+
+  constructor (key: string, kvs: KVS, logger: SDKLogger.Logger, pubSub: PubSub, thirdpartyRequests: ThirdpartyRequests) {
+    this.key = key
+    this.kvs = kvs
+    this.logger = logger
+    this.pubSub = pubSub
+    this.thirdpartyRequests = thirdpartyRequests
+  }
+
+  channelName (args: OTPValidateModelArgs): string {
+    const tokens = [this.modelName, args.consentRequestId]
+    return tokens.map(x => `${x}`).join('-')
+  }
+
+  async requestAction (args: OTPValidateModelArgs): Promise<void> {
+    if (!args.fspId) {
+      throw new Error('OTPValidate args requires \'fspId\' to be nonempty string')
+    }
+
+    if (!(args.consentRequest && typeof (args.consentRequest) === 'object')) {
+      throw new Error('OTPValidate.requestAction args requires \'transfer\' to be specified')
+    }
+    this.thirdpartyRequests.patchConsentRequests(args.consentRequestId, args.consentRequest, args.fspId)
+    resolve()
```

Comment:
let never use `path.resolve` when Promise<void> is expected. It is an antipattern and brings not-needed dependency and note expected side effects: https://nodejs.org/api/path.html#path_path_resolve_paths

When async function doesn't return any value it will be transformed into `Promise<void>` according to its signature - but `await` expression in the body should be present.

If you would like to be explicit then use promise factory: `return Promise.resolve()`